### DISCUSSION TITLE
若干参照 Z-Library 的新功能建议

### DISCUSSION THREAD

**Discussion by NoNormalCreeper** [byrdocs/byrdocs-archive#3] [Ideas]
1. 预览 PDF 文件，用户在下载前即可预览欲下载的书籍/资料，提前检查是否是想要的资料。
2. 多格式、多渠道下载。例如下载时可以选择转换为其他格式下载（如 epub 等），也可以选择通过浏览器下载、通过邮箱发送等。
3. 高级搜索：能通过学院、年度等标签筛选，通过不同规则（如相关程度、下载量。上传时间等）对搜索结果进行排序。

**Comment by cppHusky** [comment]
你好，

经过商讨，我们认为：
- 高级搜索是比较紧急，也比较重要的需求，我们计划在明年2月前完成这个功能。
- 预览功能可以做，但目前来说意义有限（对于书籍和试题而言，整份文件的大部分信息都可以通过封面页知道）。我们会考虑较晚实现这个功能。
- 邮箱发送的意义不大，况且有 25MB 的大小限制，不值得做。
- 现阶段我们不打算支持 PDF 以外的其它格式。出于标准化/去重/封面解析考虑，PDF 是我们的最佳选择。
- 至于 EPUB 转换，其实对于大部分电子书来说都不可行，我们也不会专门去做这个功能。

总之，感谢你的建议。

**Reply by NoNormalCreeper** [reply]
了解，非常感谢您的回复。

另外，注意到很多地方的预览功能似乎都是通过外部库实现的，即服务器后端传入文件，便可实时渲染出一个页面来，这或许会使开发容易许多（无需重复造轮子）。以此提供一个有可能的解决方案。

**Comment by YouXam** [comment]
pdf 预览功能已完成

**Comment by YouXam** [comment]
高级搜索中的筛选功能已完成，排序功能暂不考虑
